### PR TITLE
Add button group component

### PR DESCRIPTION
**Pull request by saralohr**
button-group

### CODE REVIEW

File: src/components/Button.scss [hunk 1]
```diff
@@ -50,8 +50,6 @@ $Button-padding: 0 scaleGrid(2);
 
   &.is-active {
     background-color: $Button-bg-hover;
-    border-bottom-width: 1px;
-    border-top-width: $Button-border-bottom-width;
   }
```
Comment (saralohr) [inline src/components/Button.scss]: @williamdingwall What do you think about this change? Adding the heavy border to the top on active state always seemed a little weird to me... especially in the button group, but maybe that's just me!

File: src/components/ButtonGroup.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+.ButtonGroup {
+  display: flex; // 1
+  display: -ms-flexbox; // 2
+}
+
+
+// * 1. Overlap elements and override any vertical margins.
+// * 2. Reset `border-radius` so corners line up.
+
+.ButtonGroup > * {
```
Comment (williamdingwall) [inline src/components/ButtonGroup.scss]: Couldn't this just be .ButtonGroup > .Button, I'm guessing the button class will always be the direct children of button right?

File: src/components/ButtonGroup.scss [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  margin: 0 $ButtonGroup-space; // 1
+  border-radius: 0; // 2
+}
+
+
+// * Apply border-radius outer edges of first and last children
+
+.ButtonGroup > *:first-child {
```
Comment (williamdingwall) [inline src/components/ButtonGroup.scss]: Same here

File: src/components/ButtonGroup.scss [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  border-top-left-radius: $ButtonGroup-border-radius;
+  border-bottom-left-radius: $ButtonGroup-border-radius;
+}
+
+.ButtonGroup > *:last-child {
```
Comment (williamdingwall) [inline src/components/ButtonGroup.scss]: and here

File: src/components/ButtonGroup.scss (hunk 2 above)
Comment (saralohr) [inline src/components/ButtonGroup.scss]: Yep, prolly so! I'll change this.

**Pull request merged**